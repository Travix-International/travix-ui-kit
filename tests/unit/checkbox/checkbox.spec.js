import * as enzyme from 'enzyme';
import * as enzymeToJson from 'enzyme-to-json';
import React from 'react';
import Checkbox from '../../../components/checkbox/checkbox';

describe('Checkbox', () => {
  describe('#render()', () => {
    const shallow = enzyme.shallow;
    const shallowToJson = enzymeToJson.shallowToJson;
    const onChange = () => {};

    it('should render disabled checkbox', () => {
      const wrapper = shallow(
        <Checkbox
          disabled
          name="disabled"
          onChange={onChange}
        />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render checked checkbox', () => {
      const wrapper = shallow(
        <Checkbox
          checked
          name="checked"
          onChange={onChange}
        />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render checkbox', () => {
      const wrapper = shallow(
        <Checkbox
          name="checkbox"
          onChange={onChange}
        />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
