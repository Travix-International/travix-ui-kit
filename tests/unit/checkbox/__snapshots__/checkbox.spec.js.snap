// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox #render() should not modify mods 1`] = `
<label
  className="ui-checkbox ui-checkbox_test"
  htmlFor=""
>
  <input
    aria-checked={false}
    checked={false}
    disabled={false}
    id=""
    role="radio"
    type="checkbox"
  />
  <span
    className="ui-checkbox__text"
  />
</label>
`;

exports[`Checkbox #render() should render checkbox 1`] = `
<label
  className="ui-checkbox"
  htmlFor="checkbox"
>
  <input
    aria-checked={false}
    checked={false}
    disabled={false}
    id="checkbox"
    onChange={[Function]}
    role="radio"
    type="checkbox"
  />
  <span
    className="ui-checkbox__text"
  />
</label>
`;

exports[`Checkbox #render() should render checked checkbox 1`] = `
<label
  className="ui-checkbox"
  htmlFor="checked"
>
  <input
    aria-checked={true}
    checked={true}
    disabled={false}
    id="checked"
    onChange={[Function]}
    role="radio"
    type="checkbox"
  />
  <span
    className="ui-checkbox__text"
  />
</label>
`;

exports[`Checkbox #render() should render disabled checkbox 1`] = `
<label
  className="ui-checkbox ui-checkbox_is-disabled"
  htmlFor="disabled"
>
  <input
    aria-checked={false}
    checked={false}
    disabled={true}
    id="disabled"
    onChange={[Function]}
    role="radio"
    type="checkbox"
  />
  <span
    className="ui-checkbox__text"
  />
</label>
`;
